### ISSUE TITLE
The suggested code by `option_map_or_none` is warned again in some case

### ISSUE THREAD

**Issue by giraffate** [rust-lang/rust-clippy#7960]
```rs
fn main() {
    let opt = Some(3);
    opt.map_or(None, |a| Some(a + 1));
}
```
Clippy warns like below on this:
```
  |
3 |     opt.map_or(None, |a| Some(a + 1));
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try using `and_then` instead: `opt.and_then(|a| Some(a + 1))`
  |
```
So I updated it according to the warning:
```rs
fn main() {
    let opt = Some(3);
    opt.and_then(|a| Some(a + 1));
}
```
But, Clippy warns like below again:
```
  |
3 |     opt.and_then(|a| Some(a + 1));
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `opt.map(|a| a + 1)`
  |
```

In this case, I want `option_map_or_none` to suggest the last code suggested from the beginning.
